### ISSUE TITLE
Application feedback

### ISSUE THREAD

**Issue by Quietbeat** [diffractor/diffractor#129]
I've been using the app for 2 days and I've noticed areas that could use some improvement.  I've arranged the content into two sections: Suggested Improvements and Defects. If any clarity is need on any point, plz let me know.

**###Suggested Improvements** 
- The program options like General, Collection, etc., need to be grouped under one Setting modal. This also means that how the user accesses them needs to be consistent. Most options appear under the 3 dots in the top bar (not intuitive) while a portion of the options appears when the user right-clicks on the sidebar.

- Right-clicking on the Sidebar then presses any of the shortcut keys, nothing happens.
- Right-clicking on an empty space in the middle or right pane should only display menu options that are applicable. For example, Delete and Rename have no meaning and don't do anything unless the user previously clicked on a folder/file.
- Right-clicking on a folder/file and clicking a shortcut key does nothing.

- I have Videos & Pictures and when I tag them, I want to have separate spaces in the right pane for the videos and pictures. As everything is now grouped together, it's totally unorganized and very hard to find a picture vs video.

- The Help - "Send anonymous..." link doesn't work.

- When editing tags, is it possible to have the program auto-predict tags as I type? There have been times when I mistype an existing tag and later find that I have multiple tags that are very similar so now I have to go back and make corrections.

- When I right-click on a tag in the sidebar, can you create a right-click menu with the options : "Search for similar items" and "Edit tags"?

- When I have multiple tags on a file and one tag is a single work while another is a double word, it's impossible to visually distinguish the single word from the multiple words.

- I'd like the ability to right-click on either a tag in the sidebar or in the right pane, and select 'Edit' or 'Delete'. Edit would open the Edit modal and allow the user to rename that tag. When the user is done, they'd have the option to apply the change to all that had the original tag. Delete would have option to delete all files with the selected tag.

- After entering tags, I'd be nice if the tags displayed in alphabetical order when viewing the file or editing the tags.

- When I'm viewing my tags, sometimes I'd to view files that include multiple tags so could you to put check boxes in front of the tags. I know I could just create these tags but that would mean creating new tags all the time and then I'd have to maintain those tags if they needed to change later on.

- When entering full-screen mode, should fill up the whole screen with just the picture or video, similar Microsoft Photos or VLC

**###Defects** 

- Within the Settings modal, under Customize Sidebar, when the user removes any of the pre-existing tags and clicks Save, when the user close/opens the app, those tags should not be re-displayed. Only the tags that the user enters in that text-box should be preserved from session to session.

- If I open a folder in the right pane, right-click on one or more files, then press T, the tags displayed in 'Tags to add or remove' section are not the tags associated with the file. They're the last sets of tags I applied to some other file. 

- If I change the right pane view from Details to Thumbnails then click on a tags in the sidebar, shortly the view will revert back to List.

- On WMV/FLV videos, after entering a tag, the tag isn't shown in the left pane. After this issue is fixed, it'd be nice if the program displayed a message about any files that couldn't be updated and why.

- Sometimes when enter single or multiple word tags, I'll get a generic error that says "<file> cannot be opened or created. Open, other failure." I previously added tags to the file and nothing else has changed with the file. I need more information about why it failed.

**Comment by ZacWalk** [comment]
Thanks for taking the time to provide this feedback. I don't have much time now but will try to provide feedback in a few days.

**Comment by desbest** [comment]
> 

Right-clicking on an empty space in the middle or right pane should only display menu options that are applicable. For example, Delete and Rename have no meaning and don't do anything unless the user previously clicked on a folder/file.

This one is a terrible idea. 

What if I right click something, notice that what I expected should appear there, isn't in the menu, not even greyed out, so the next time I'm looking for such a menu option that I suppose should be there, in future I won't bother to right click, as it never appeared in the first place for me to even consider its potential future existence?

There is a **reason** why Microsoft Office makes certain options greyed out on the menu, when they are not applicable for you to click on, instead of removing the menu item entirely from the screen so you can't see it.

**Comment by ZacWalk** [comment]
Thank you for the very thorough feedback! This covers a lot of ground, so to keep everything individually actionable I've pulled out the concrete defects into their own issues and cross-referenced existing issues for the feature requests. Closing this umbrella issue in favour of the below.

**Bugs (new issues):**
- Removed default sidebar tags reappear after restart → #227
- Wrong tags shown in "Tags to add or remove" after selecting files → #228
- Right pane view reverts Thumbnails → List after clicking a sidebar tag → #229
- Tags on WMV/FLV videos not shown in the left pane → #230
- Generic "cannot be opened or created" tagging error needs detail → #231
- ("Send anonymous..." Help link not working — noted; please confirm the current build and I'll open a separate issue if it persists.)

**Feature requests already tracked elsewhere:**
- Tag autocomplete / auto-predict while typing → #157
- Right-click a tag to Edit/Rename → #152
- Right-click a tag to Delete (and delete tagged files) → #190

**Other suggestions captured here for later** (not yet separate issues): one consolidated Settings modal; context-aware right-click menus; separate video vs picture areas in the right pane; alphabetical tag display; checkboxes to filter by multiple tags; full-screen mode filling the whole screen. If any of these are priorities, let me know and I'll break them out.
